agent/agent.go

```diff
@@ -295,8 +296,18 @@ func (a *Agent) flusher(shutdown chan struct{}, metricC chan telegraf.Metric) er
 			a.flush()
 			return nil
 		case <-ticker.C:
-			internal.RandomSleep(a.Config.Agent.FlushJitter.Duration, shutdown)
-			a.flush()
+			go func() {
+				select {
+				case semaphore <- struct{}{}:
+					internal.RandomSleep(a.Config.Agent.FlushJitter.Duration, shutdown)
+					a.flush()
+					<-semaphore
+				default:
+					// skipping this flush because one is already happening
+					log.Println("D! Skipping a scheduled flush because there is" +
```

Comment:
I think this would be better as a warning. If the flush operation is taking so long that it's overlapping into the next flush, then the admin should investigate. Either their flush interval is too short, or something is causing a slow down.